src/context/sessionContext/index.tsx

```diff
@@ -104,8 +107,13 @@ export const SessionProvider = ({
   if (typeof window !== "undefined") {
     currentLocation = window.location;
   }
-
   useEffect(() => {
+    if (
+      typeof restoreSession === "undefined" &&
+      typeof onSessionRestore !== "undefined"
+    ) {
+      return;
+    }
```

Comment:
If I understand this correctly, you're doing this because when an `onSessionRestore` prop is passed, `restoreSession` will not be set until `setRestoreSession` is called.

However, there's not really a reason to keep `restoreSession` in React state, is there? As in, can't you derive it directly from the prop outside of a `useEffect` using `const restoreSession = typeof onSessionRestore !== "undefined"`?